library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/server/auth/ExampleGafferPopAuthenticator.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.tinkerpop.server.auth;
+
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticatedUser;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticationException;
+import org.apache.tinkerpop.gremlin.server.auth.Authenticator;
+
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An example authenticator class for GafferPop, this should not
+ * be used in production as it allows all user and password combinations.
+ * The class is intended as a template for an deployment specific class
+ * that hooks into a proper authorisation mechanism such as LDAP etc.
+ */
+public class ExampleGafferPopAuthenticator implements Authenticator {
+
+    @Override
+    public boolean requireAuthentication() {
+        return true;
+    }
+
+    @Override
+    public void setup(final Map<String, Object> config) {
+        // Nothing to do
+    }
+
+    @Override
+    public SaslNegotiator newSaslNegotiator(final InetAddress remoteAddress) {
+        return new PlainTextSaslAuthenticator();
+    }
+
+
+    @Override
+    public AuthenticatedUser authenticate(final Map<String, String> credentials) throws AuthenticationException {
+        // Get the username from the credentials set by the SASL negotiator
+        final String username = credentials.get("username");
+        return new AuthenticatedUser(username);
+    }
+
+    /**
+     * Very simple SASL authenticator that will just extract username and password
+     * from plain text.
+     */
+    private class PlainTextSaslAuthenticator implements Authenticator.SaslNegotiator {
+        private static final byte NUL = 0;
+        private boolean complete = false;
+        private String username;
+        private String password;
+
+        @Override
+        public byte[] evaluateResponse(final byte[] clientResponse) throws AuthenticationException {
+            decodeCredentials(clientResponse);
+            complete = true;
+            return new byte[0];
+        }
+
+        @Override
+        public boolean isComplete() {
+            return complete;
+        }
+
+        @Override
+        public AuthenticatedUser getAuthenticatedUser() throws AuthenticationException {
+            if (!complete) {
+                throw new AuthenticationException("SASL negotiation not complete");
+            }
+            final Map<String, String> credentials = new HashMap<>();
+            credentials.put("username", username);
+            credentials.put("password", password);
```

Comment:
Just thinking will this be plain text in memory? If so, just wondering how we would go about ensuring that this is secure through the journey of the authentication?

Reply:
As mentioned in previous comment and in the javadocs this is not a production ready class it is just an example for plain text authentication for us to test with each deployment will have to implement their own `Authenticator` that hooks into their own authentication method e.g. LDAP, Kerberos etc.